agent_plugin_builder/vendor_dirs.py

```diff
@@ -1,54 +1,81 @@
+WINDOWS_BUILD_PACKAGE_LIST_COMMANDS: Final = " && ".join(
+    [
+        WINDOWS_IMAGE_INIT_COMMAND,
+        "cd /plugin",
+        "wine pip install --dry-run -r requirements.txt --report windows.json",
+        "chown {uid}:{gid} /plugin/windows.json",
+    ]
+)
+WINDOWS_BUILD_VENDOR_DIR_COMMANDS: Final = " && ".join(
+    [
+        WINDOWS_IMAGE_INIT_COMMAND,
+        "cd /plugin",
+        "wine pip install -r requirements.txt -t src/vendor-windows",
+        "chown -R {uid}:{gid} /plugin/src/vendor-windows",
+    ]
+)
 
 
-def check_if_common_vendor_dir_possible(build_dir: Path, uid, gid) -> bool:
-    requirements_file = build_dir / "requirements.txt"
-    command = ["pipenv", "requirements"]
-    with requirements_file.open("w") as f:
-        subprocess.check_call(command, cwd=str(build_dir), stdout=f)
+def check_if_common_vendor_dir_possible(build_dir: Path, uid: int, gid: int) -> bool:
+    """
+    Check if a common vendor directory is possible by comparing the package lists generated
+    from a dry run of the requirements installation on Linux and Windows.
 
-    client = docker.from_env()
-    linux_commands = [
-        'export PYENV_ROOT="$HOME/.pyenv"',
-        'command -v pyenv >/dev/null || export PATH="$PYENV_ROOT/bin:$PATH"',
-        'eval "$(pyenv init -)"',
-        "cd /plugin && pip install --dry-run -r requirements.txt --report linux.json",
-        f"chown -R {uid}:{gid} /plugin/linux.json",
-    ]
-    windows_commands = (
-        ". /opt/mkuserwineprefix && "
-        "cd /plugin && wine pip install --dry-run -r requirements.txt --report "
-        f"windows.json && chown {uid}:{gid} /plugin/windows.json"
-    )
+    :param build_dir: Path to the build directory.
+    :param uid: User ID to set on the vendor directory.
+    :param gid: Group ID to set on the vendor directory.
+    """
+    generate_requirements_file(build_dir)
 
-    full_linux_command = "/bin/bash -c '" + " && ".join(linux_commands) + "'"
-    client.containers.run(
-        AGENT_PLUGIN_IMAGE,
-        command=full_linux_command,
-        volumes={build_dir: {"bind": "/plugin", "mode": "rw"}},
-        remove=True,
+    command = _build_bash_command(
+        LINUX_BUILD_PACKAGE_LIST_COMMANDS.format(uid=quote(str(uid)), gid=quote(str(gid)))
     )
-    client.containers.run(
-        PLUGIN_BUILDER_IMAGE,
-        command=f'/bin/bash -c "{windows_commands}"',
-        volumes={build_dir: {"bind": "/plugin", "mode": "rw"}},
-        remove=True,
+    linux_container = _run_container_with_plugin_dir(AGENT_PLUGIN_IMAGE, command, build_dir)
+    _log_container_output(linux_container, "Linux Dry Run requirements, ")
+
+    command = _build_bash_command(
+        WINDOWS_BUILD_PACKAGE_LIST_COMMANDS.format(uid=quote(str(uid)), gid=quote(str(gid)))
     )
+    windows_container = _run_container_with_plugin_dir(PLUGIN_BUILDER_IMAGE, command, build_dir)
+    _log_container_output(windows_container, "Windows Dry Run requirements, ")
```

Comment:
It would probably be better if the "prefix" was a string that didn't assume some kind of formatting. In other words, `_log_container_output()` should be responsible for handling commas, spacing, whatever.